homeassistant/components/knx/__init__.py

```diff
@@ -276,61 +331,47 @@ def __init__(self, hass: HomeAssistant, config: ConfigType) -> None:
     def init_xknx(self) -> None:
         """Initialize XKNX object."""
         self.xknx = XKNX(
-            own_address=self.config[DOMAIN][CONF_KNX_INDIVIDUAL_ADDRESS],
-            rate_limit=self.config[DOMAIN][ConnectionSchema.CONF_KNX_RATE_LIMIT],
-            multicast_group=self.config[DOMAIN][ConnectionSchema.CONF_KNX_MCAST_GRP],
-            multicast_port=self.config[DOMAIN][ConnectionSchema.CONF_KNX_MCAST_PORT],
+            own_address=self.config[CONF_KNX_INDIVIDUAL_ADDRESS],
+            rate_limit=self.config[ConnectionSchema.CONF_KNX_RATE_LIMIT],
+            multicast_group=self.config[ConnectionSchema.CONF_KNX_MCAST_GRP],
+            multicast_port=self.config[ConnectionSchema.CONF_KNX_MCAST_PORT],
             connection_config=self.connection_config(),
-            state_updater=self.config[DOMAIN][ConnectionSchema.CONF_KNX_STATE_UPDATER],
+            state_updater=self.config[ConnectionSchema.CONF_KNX_STATE_UPDATER],
         )
 
     async def start(self) -> None:
         """Start XKNX object. Connect to tunneling or Routing device."""
         await self.xknx.start()
-        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self.stop)
+        self.hass_remove_listener = self.hass.bus.async_listen_once(
```

Comment:
I would just attach them to the config entry: `entry.async_on_unload(self.hass.bus.async_listen_once(…))`